graph/loaders/comments.js

```diff
@@ -1,85 +1,134 @@
-const DataLoader = require('dataloader');
-
 const util = require('./util');
 
-const ActionModel = require('../../models/action');
 const CommentModel = require('../../models/comment');
-const CommentsService = require('../../services/comments');
 
 /**
- * Retrieves comments by an array of asset id's, results are returned in reverse
- * chronological order.
- * @param {Array} ids array of ids to lookup
+ * Returns the comment count for all comments that are public based on their
+ * asset ids.
+ * @param {Object}        context     graph context
+ * @param {Array<String>} asset_ids   the ids of assets for which there are
+ *                                    comments that we want to get
  */
-const genCommentsByAssetID = (context, ids) => {
-  return CommentModel.find({
-    asset_id: {
-      $in: ids
+const getCountsByAssetID = (context, asset_ids) => {
+  return CommentModel.aggregate([
+    {
+      $match: {
+        asset_id: {
+          $in: asset_ids
+        },
+        status: {
+          $in: [null, 'ACCEPTED']
```

Comment:
This works, but in the future we may want to abstract the concept of 'publicly view-able comments' into a function that returns or a constant that holds this array. I can see this array being hard coded in several places.